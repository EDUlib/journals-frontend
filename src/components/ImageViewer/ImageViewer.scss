// TODO: all colors should be using variables and inheriting from sitebranding
.image-border{
  background-color: #f2f8fb;
}
.image-title{
  padding: 4px;
  font-size: 14px;
  font-weight: 600;
  font-style: normal;
  font-stretch: normal;
  line-height: 1.71;
  letter-spacing: 1px;
  color: #292b2c;
}
.image-block{
  max-width: 100%;
  margin-left: auto;
  margin-right: auto;
  padding: 4px;
  display: block;
}
.image-caption{
  padding: 4px;
  font-size: 14px;
  font-weight: normal;
  font-style: normal;
  font-stretch: normal;
  line-height: 1.5;
  letter-spacing: normal;
  color: #292b2c;
}
