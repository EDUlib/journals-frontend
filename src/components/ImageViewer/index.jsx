import React from 'react';
import PropTypes from 'prop-types';

// TODO: get from site branding
import settings from '../../data/configuration/constants';

import './ImageViewer.scss';


const ImageViewer = props => (
  // If title or caption is an empty string or a string of whitespaces do not display an empty div
  <span id={props.id}>
    <div className="image-border w-100">
      { (props.title.trim() !== '') ? <div className="image-title">{props.title} </div> : '' }
      <img className="image-block" src={`${settings.journalsBackendBaseUrl}${props.url}`} alt={props.altText} />
      { (props.caption.trim() !== '') ? <div className="image-caption">{props.caption} </div> : '' }
    </div>
  </span>
);

ImageViewer.defaultProps = {
  altText: '',
  caption: '',
  title: '',
};

ImageViewer.propTypes = {
  url: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  altText: PropTypes.string,
  caption: PropTypes.string,
  title: PropTypes.string,
};

export default ImageViewer;
