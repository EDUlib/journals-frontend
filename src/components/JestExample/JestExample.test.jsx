import { mount } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';
import Link from './Link.react';
import MySum from '.';
import ImageViewer from '../ImageViewer';


it('Image viewer with valid img', () => {
  const tree = renderer
    .create(<ImageViewer url="http://foo.com/myimg.png" altText="My Test Image" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Image viewer with missing img', () => {
  const tree = renderer
    .create(<ImageViewer url="" altText="My Test Image" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Image viewer with missing image check DOM', () => {
  const wrapper = mount((<ImageViewer url="" altText="My Test Image" />));
  expect(wrapper.find('.alert-dialog').exists()).toBeTruthy();
});

it('renders correctly', () => {
  const tree = renderer
    .create(<Link href="http://www.facebook.com">Facebook</Link>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('adds 1 + 2 to equal 3', () => {
  expect(MySum(1, 2)).toBe(3);
});

test('there is no I in team', () => {
  expect('team').not.toMatch(/I/);
});

test('but there is a "stop" in Christoph', () => {
  expect('Christoph').toMatch(/stop/);
});

test('null', () => {
  const n = null;
  expect(n).toBeNull();
  expect(n).toBeDefined();
  expect(n).not.toBeUndefined();
  expect(n).not.toBeTruthy();
  expect(n).toBeFalsy();
});

test('zero', () => {
  const z = 0;
  expect(z).not.toBeNull();
  expect(z).toBeDefined();
  expect(z).not.toBeUndefined();
  expect(z).not.toBeTruthy();
  expect(z).toBeFalsy();
});
