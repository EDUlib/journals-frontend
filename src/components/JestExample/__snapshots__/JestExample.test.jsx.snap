// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image viewer with missing img 1`] = `
<div
  className="alert fade alert-danger show"
  hidden={false}
  role="alert"
>
  <div
    className="alert-dialog"
  >
    <div
      className="alert-dialog"
    >
      <div
        className="icon mr-2"
      >
        <span
          aria-hidden={true}
          className="fa fa-times-circle"
          id="Icon1"
        />
      </div>
      <div
        className="message"
      >
        <span>
          Image not found
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Image viewer with valid img 1`] = `
<img
  alt="My Test Image"
  className="image-block"
  src="http://foo.com/myimg.png"
/>
`;

exports[`renders correctly 1`] = `
<a
  className="normal"
  href="http://www.facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Facebook
</a>
`;
